### DISCUSSION TITLE
关于使用java开发插件

### DISCUSSION THREAD

**Discussion by NanyuZhang0812** [Eiriksgata/mirai-rulateday-dice#60] [Q&A] [answered]
我想试着开发一些简单的插件，例如自动回复消息什么的，但好像找不到使用java开发的模板

**Comment by eiriksgata** [comment, marked as answer]
本项目就是纯Java开发的，不过目前功能完成度比较高，理解起来比较困难。实际上整合成插件的代码占比非常少，如果你有需要的话我可以单独整理一个开发模板出来。

**Reply by NanyuZhang0812** [reply]
如果可以的话非常感谢！

**Reply by eiriksgata** [reply]
可能没那么快，最快的话也要等到明天才行。

**Reply by NanyuZhang0812** [reply]
感谢

**Reply by eiriksgata** [reply]
你看这个项目：https://github.com/Eiriksgata/mirai-plug-examples-maven

**Reply by eiriksgata** [reply]
有什么问题可以问我

**Reply by NanyuZhang0812** [reply]
感谢orz。我现在大二，之前在搞acm，所以很多实际开发中的知识都不知道...

**Reply by eiriksgata** [reply]
那你可能还需要学习一下Maven，等其他东西。对于语法体系不熟悉的人来说，比较难开发。

**Reply by NanyuZhang0812** [reply]
关于各种事件监听的写法有没有提供api文档呢...

**Reply by eiriksgata** [reply]
可以去https://github.com/mamoe/mirai/blob/dev/docs/CoreAPI.md 看看，或者是通过官方Mirai的IDEA插件来辅助使用。

**Reply by eiriksgata** [reply]
对于我来说，我前期是直接看他们的源码了解的API，而且再加上代码追踪+常规代码设计体系，就可以知道每个方法的使用了。所以我都没怎么看过文档。

**Reply by NanyuZhang0812** [reply]
似乎也只能看源码呢...没有找到详细的api

**Reply by eiriksgata** [reply]
你可以去官方Issue求助看看

**Reply by NanyuZhang0812** [reply]
https://github.com/mamoe/mirai/tree/dev/mirai-core-api/src/commonMain/kotlin
好像这部分的源码有一些帮助
